File: 2312-selling-pieces-of-wood/2312-selling-pieces-of-wood.cpp
class Solution {
public:
    long long sellingWood(int m, int n, vector<vector<int>>& prices) {
        vector dp(m+1, vector(n+1, 0LL));
        for (auto e : prices) {
            dp[e[0]][e[1]] = e[2];
        }
        for (int i = 1; i <= m; ++i) {
            for (int j = 1; j <= n; ++j) {
                for (int x = 1; x < (int)i/2+1; ++x) {
                    dp[i][j] = max(dp[i][j], dp[x][j] + dp[i-x][j]);
                }
                for (int x = 1; x < (int)j/2+1; ++x) {
                    dp[i][j] = max(dp[i][j], dp[i][x] + dp[i][j-x]);
                }
            }
        }
        return dp[m][n];
    }
};

File: 2312-selling-pieces-of-wood/2312-selling-pieces-of-wood.py
class Solution:
    def sellingWood(self, m: int, n: int, pr: List[List[int]]) -> int:
        dp=[[0for i in range(n+1)]for j in range(m+1)]
        for r,c,p in pr:
            dp[r][c]=p
        for i in range(1,m+1):
            for j in range(1,n+1):
                for r in range(1,i//2+1):
                    dp[i][j]=max(dp[i][j],dp[r][j]+dp[i-r][j])
                for c in range(1,j//2+1):
                    dp[i][j]=max(dp[i][j],dp[i][c]+dp[i][j-c])
        return dp[-1][-1]
                    